src/acceptance-test/java/de/zalando/aruha/nakadi/repository/db/SubscriptionDbRepositoryTest.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    @Test
+    public void whenCreateSubscriptionThenOk() throws Exception {
+
+        final Subscription subscription = createSubscription();
+        repository.saveSubscription(subscription);
+
+        final int rows = template.queryForObject("SELECT count(*) FROM zn_data.subscription", Integer.class);
+        assertThat("Number of rows should be 1", rows, equalTo(1));
+
+        final Map<String, Object> result =
+                template.queryForMap("SELECT s_id, s_subscription_object FROM zn_data.subscription");
+        assertThat("Id is persisted", result.get("s_id"), equalTo(subscription.getId()));
+
+        final ObjectMapper mapper = (new JsonConfig()).jacksonObjectMapper();
+        final Subscription saved = mapper.readValue(result.get("s_subscription_object").toString(), Subscription.class);
+        assertThat("Saved subscription equal to original one", saved, equalTo(subscription));
+    }
+
+    @Test(expected = DuplicatedSubscriptionException.class)
+    public void whenCreateSubscriptionWithDuplicatedKeyParamsThenDuplicatedSubscriptionException() throws Exception {
+
+        final Subscription subscription = createSubscription();
+        repository.saveSubscription(subscription);
+
+        // set another id to subscription but keep the same key properties
+        subscription.setId(randomUUID().toString());
+        repository.saveSubscription(subscription);
+    }
+
+    @Test
+    public void whenGetSubscriptionByIdThenOk() throws InternalNakadiException, JsonProcessingException,
+            NoSuchSubscriptionException {
+
+        // insert subscription into DB
+        final Subscription subscription = createSubscription();
+        insertSubscriptionToDB(subscription);
+
+        // get subscription by id and compare to original
+        final Subscription gotSubscription = repository.getSubscription(subscription.getId());
+        assertThat("We found the needed subscription", gotSubscription, equalTo(subscription));
+    }
+
+    @Test
+    public void whenGetSubscriptionByKeyPropertiesThenOk() throws InternalNakadiException, JsonProcessingException,
+            NoSuchSubscriptionException {
+
+        // insert subscription into DB
+        final Subscription subscription = createSubscription("myapp", ImmutableSet.of("my-et", "second-et"), "mycase");
+        insertSubscriptionToDB(subscription);
+
+        // get subscription by key properties and compare to original
+        final Subscription gotSubscription = repository.getSubscription("myapp", ImmutableSet.of("second-et", "my-et"),
+                "mycase");
+        assertThat("We found the needed subscription", gotSubscription, equalTo(subscription));
+    }
+
+    private void insertSubscriptionToDB(final Subscription subscription) throws JsonProcessingException {
+        template.update("INSERT INTO zn_data.subscription (s_id, s_subscription_object) VALUES (?, ?::jsonb)",
```

Comment:
It's a bootstrap for test. It should not use any of the project's code (because the project code is actually what we test).